packages/smaz-decompress/src/index.ts

```diff
@@ -26,3 +26,54 @@ export class SmazDecompress {
     return output;
   }
 }
+
+export class SmazDecompressRaw {
+  public static fromStringCodebook(codebook: readonly string[]) {
+    return new this(
+      codebook.map((str) => {
+        const arr = new Uint8Array(str.length);
```

Comment:
Could a codebook contain unicode characters?

Reply:
Nice catch, we may fail at that point:

```sh
Welcome to Node.js v22.14.0.
Type ".help" for more information.
> const a = '🥳'
undefined
> a
'🥳'
> a.length
2
> const b = new Uint8Array(2)
undefined
> b[0] = a[0]
'\ud83e'
> b[1] = a[1]
'\udd73'
> b
Uint8Array(2) [ 0, 0 ]
```

They can live in codebook but they need to be properly encoded.

Reply:
Handled by 4f86cdf39b9d075cdc9070a5c7b6f9a6e689e60f

Reply:
I believe sticking with TextEncoder here is the best:

1. `TextEncoder` is widely available
2. UTF8 is commonly used and fully compatible with ASCII
3. We don't guarantee other encodings and we cannot encode them natively

refs [`5b0896a` (#841)](https://github.com/remusao/mono/pull/841/commits/5b0896a0630753a008dc9383d62b6572200b48ea)